File: frameworks/shared/ruby/lib/velocitybench/healthcheck/health_check.rb
# frozen_string_literal: true

module VelocityBench
  module HealthCheck
    # Individual health check result.
    class HealthCheck
      attr_accessor :status
      attr_reader :response_time_ms, :error, :warning, :info, :additional_data

      def initialize(status)
        @status = status
        @response_time_ms = nil
        @error = nil
        @warning = nil
        @info = nil
        @additional_data = {}
      end

      def with_response_time(ms)
        @response_time_ms = ms
        self
      end

      def with_error(error)
        @error = error
        self
      end

      def with_warning(warning)
        @warning = warning
        self
      end

      def with_info(info)
        @info = info
        self
      end

      def with_data(key, value)
        @additional_data[key] = value
        self
      end

      def to_h
        result = { status: @status }
        result[:response_time_ms] = @response_time_ms if @response_time_ms
        result[:error] = @error if @error
        result[:warning] = @warning if @warning
        result[:info] = @info if @info
        result.merge(@additional_data)
      end

      def to_json(*_args)
        to_h.to_json
      end
    end
  end
end

File: frameworks/shared/java/src/main/java/com/velocitybench/healthcheck/HealthCheck.java
package com.velocitybench.healthcheck;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.HashMap;
import java.util.Map;

/**
 * Individual health check result.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
public class HealthCheck {
    private HealthStatus status;

    @JsonProperty("response_time_ms")
    private Double responseTimeMs;

    private String error;
    private String warning;
    private String info;

    private Map<String, Object> additionalData;

    public HealthCheck(HealthStatus status) {
        this.status = status;
        this.additionalData = new HashMap<>();
    }

    public HealthStatus getStatus() {
        return status;
    }

    public void setStatus(HealthStatus status) {
        this.status = status;
    }

    public Double getResponseTimeMs() {
        return responseTimeMs;
    }

    public HealthCheck withResponseTime(double ms) {
        this.responseTimeMs = ms;
        return this;
    }

    public String getError() {
        return error;
    }

    public HealthCheck withError(String error) {
        this.error = error;
        return this;
    }

    public String getWarning() {
        return warning;
    }

    public HealthCheck withWarning(String warning) {
        this.warning = warning;
        return this;
    }

    public String getInfo() {
        return info;
    }

    public HealthCheck withInfo(String info) {
        this.info = info;
        return this;
    }

    public Map<String, Object> getAdditionalData() {
        return additionalData;
    }

    public HealthCheck withData(String key, Object value) {
        this.additionalData.put(key, value);
        return this;
    }
}
